### PR TITLE
Gleam definitions v3

### PR DESCRIPTION
**Pull request by wolverdude**
## Overview

Addresses [Covid#444](https://github.com/epidemics/covid/issues/444). This PR deprecates the Gleam Definitions colab in favor of an epimodel task that can pull data from a Google Sheet or a CSV with groups and traces configured in YAML.

This changes `config.yaml` somewhat, in that the `scenarios`/`groups` config has been renamed and all placed under a single master key `scenarios`. The new format can be seen in the [test config](https://github.com/epidemics/epimodel/blob/wolverdude/gleam-definitions-v3/tests/data/scenario/config.yaml). Instructions for how the scenarios differ numerically no longer live in `config.yaml`, but are instead taken from a spreadsheet like [this one](https://docs.google.com/spreadsheets/d/1IxPMadPxjnphWSKG_6PxmsrCLoXe3cHGp1Ok9kcddPk/edit#gid=1831691945).

## Code

Most of the logic is contained in [`scenario.py`](https://github.com/epidemics/epimodel/blob/wolverdude/gleam-definitions-v3/epimodel/gleam/scenario.py) with extensive unit tests in [`test_scenario.py`](https://github.com/epidemics/epimodel/blob/wolverdude/gleam-definitions-v3/tests/test_scenario.py)

#### `InputParser`

This is responsible for taking DataFrames of raw inputs and transforming them into a format that `SimulationSet` can use. This includes:

- Converting region codes or names into `Region` objects
- Distributing estimates down to the `gleam_basin` level
- Fetching distribution averages from Foretold

#### `SimulationSet`

This class decides how many simulations to create and separates the inputs to ensure each simulation is configured properly. It then passes those inputs to a set of `DefinitionBuilder`s and stores the resulting `GleamDefinition`, which it can add to a `Batch` object.

#### `DefinitionBuilder`

This wraps a `GleamDefinition` object and applies DataFrames of estimates and parameters to it. The result is a properly-formatted XML that can be fed into GLEAMviz.

#### `GleamDefinition`

More configuration options were added, and the interface was changed slightly, particularly around adding compartment estimates.

#### `Batch`

Minimal changes have been made to make integration with the `scenario` classes easier.

#### `generate_simulations()`

This function will be removed and replaced. This is the whole point of this feature.

#### `epidemics_org.py` / `web-export`

This code will be updated to handle the new configuration format.

## Still ToDo:

- [x] replace `"PLACEHOLDER"`
- [x] use `name` from `config.yaml`
- [x] use `test_definition.xml`
- [x] test integration with `Batch`
- [x] create a master run function
- [x] handle `initialCompartments`
- [x] add new dependencies
- [x] fix tests failing due to env
- [x] create luigi task
- [x] update `web-export` for new config format
- [x] get the entire flow working
- [x] use Luigi for `parameters.csv` input
- [x] update `README.md`
- solve `ergo` dependency issue (will hand off)

### CODE REVIEW

**Comment by wolverdude** [comment]
@hnykda I'm currently unable to run the luigi pipeline to test this because of a pickling issue with RegionsDataset. I would love if you could take a look at this tomorrow.

**Comment by hnykda** [comment]
Hey. Can you share more info on what exactly the problem is? E.g. posting traceback, command, ... so I can reproduce?

I just checked out your branch, run `poetry install` and it worked:
image

File: epimodel/gleam/definition.py [hunk 1]
```diff
@@ -1,36 +1,55 @@
...
 
-        self.updated = datetime.now()
+        if not self.get_timestamp_str():
+            self.set_timestamp()
 
     def copy(self):
         return copy.deepcopy(self)
 
-    def fa(self, query):
+    def find_all(self, query):
```
Comment (hnykda) [inline epimodel/gleam/definition.py:49]: very much appreciated :-D

File: epimodel/gleam/definition.py [hunk 2]
```diff
@@ -42,14 +61,94 @@ def save(self, file):
         self.tree.write(file)  # , default_namespace=self.ns['gv'])
         log.debug(f"Written Gleam definition to {file!r}")
 
+    def to_xml_string(self):
+        b = io.BytesIO()
+        self.save(b)
+        return b.getvalue().decode("ascii")
+
+    ### Testing methids
```
Comment (hnykda) [inline epimodel/gleam/definition.py]:
```suggestion
    ### Testing methods
```

**Review by hnykda — APPROVED** [review]
Code-wise, it looks very good :+1: . Great job Jon

**Comment by AcesoUnderGlass** [comment]
This breaks example-run.sh

```2020-05-25 10:54:49:luigi-interface:ERROR: [pid 13211] Worker Worker(salt=729259557, workers=1, host=Elizabeths-MBP, username=elvan, pid=13211) failed    WebUpload(gs_prefix=gs://static-covid/static/v4/deleteme, channel=test, exported_data=data-dir/outputs/example-1/web-exports/test-output)```

Still narrowing down the problem, but it might take a while due to my unfamiliarity with the new system so I wanted to register this.

**Comment by AcesoUnderGlass** [comment]
Okay, my concern now repros on master and so is not the fault of this PR.

**Pull request merged**